Upgrade ktlint plugin to 12.1.0

wolfs opened the pull request:
This version requires at least ktlint 0.47.1.

wolfs committed:
Upgrade ktlint plugin to 12.1.0

This version requires at least ktlint 0.47.1.

eskatos reviewed (changes requested):
We can't accept the file name rule change.
Moreover our custom ruleset is completely ignored by ktlint 0.47.
There are breaking changes in 0.47, more in 0.48, 0.49 and 0.50 ... and then more in 1.0 ... things look less crazy after 1.0.
The amount of work to pass the upgrade is not trivial. We need to rewrite all our custom rules.
I suggest we abandon this PR and schedule some time to either do the hard work necessary for the upgrade or reconsider this plugin entirely.

wolfs committed:
Not working: Migrate to ktlint rules v3

wolfs committed:
Revert "Not working: Migrate to ktlint rules v3"

This reverts commit 9f5c05a664f39b425d60ce19d747593ba9143766.

wolfs committed:
Fix ruleset application

- The ktlint plugin changed the name of the ruleset configuration
- There is a new base class for RuleSetProviders

wolfs committed:
Disable FilenameRule

Now requires all files to be Pascal case - we don't want that.

wolfs committed:
Polish GradleKotlinDslRuleSetProvider

wolfs committed:
Stop using deprecated `visit` method

eskatos reviewed (approved):
👍 
Good approach. This unblocks us for now until we need to upgrade ktlint further.
